linodecli/plugins/get-kubeconfig.py

```diff
@@ -0,0 +1,169 @@
+    )
+    group.add_argument(
+        "--id",
+        metavar="ID",
+        nargs="?",
+        help="ID for desired cluster. If omitted, the Label must be provided.",
+    )
+
+    parsed = parser.parse_args(args)
+
+    # If --id was used, fetch the kubeconfig using the provided id
+    if parsed.id:
+        code, kubeconfig = context.client.call_operation(
+            "lke", "kubeconfig-view", args=[parsed.id]
+        )
+
+        if code != 200:
+            print(f"Error retrieving kubeconfig: {code}", file=sys.stderr)
+            sys.exit(1)
+
+    # If --label was used, fetch the kubeconfig using the provided label
+    elif parsed.label:
+        kubeconfig = _get_kubeconfig_by_label(parsed.label, context.client)
+    else:
+        print("Either --label or --id must be used.", file=sys.stderr)
+        sys.exit(1)
+
+    # Load the specified cluster's kubeconfig and the current kubeconfig
+    clusterConfig = yaml.safe_load(
+        base64.b64decode(kubeconfig["kubeconfig"]).decode()
+    )
+    currentConfig = None
+
+    if os.path.exists(os.path.expanduser(parsed.kubeconfig)):
+        currentConfig = _load_config(os.path.expanduser(parsed.kubeconfig))
+
+    # If there is no current kubeconfig, dump the cluster config to the specified file location.
+    # If there is a current kubeconfig, merge it with the cluster's kubeconfig
+    clusterConfig = (
+        _merge_dict(currentConfig, clusterConfig)
+        if currentConfig is not None
+        else clusterConfig
+    )
+    if parsed.dry_run:
+        print(clusterConfig)
+    else:
+        _dump_config(os.path.expanduser(parsed.kubeconfig), clusterConfig)
+
+
+# Fetches the kubeconfig of the lke cluster with the specified label
+def _get_kubeconfig_by_label(cluster_label, client):
+    """
+    Returns the LKE Cluster with the given Label
+    """
+
+    code, cluster = client.call_operation(
+        "lke", "clusters-list", args=["--label", cluster_label]
+    )
+
+    if code != 200:
+        print(f"Error retrieving cluster: {code}", file=sys.stderr)
+        sys.exit(1)
+
+    if len(cluster["data"]) == 0:
+        print(
+            f"Cluster with label {cluster_label} does not exist.",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+    code, kubeconfig = client.call_operation(
+        "lke", "kubeconfig-view", args=[str(cluster["data"][0]["id"])]
+    )
+
+    if code != 200:
+        print(f"Error retrieving kubeconfig: {code}", file=sys.stderr)
+        sys.exit(1)
+
+    return kubeconfig
+
+
+# Loads a yaml file
+def _load_config(filepath):
+    with open(filepath, "r", encoding="utf-8") as file_descriptor:
+        data = yaml.load(file_descriptor, Loader=yaml.Loader)
+
+    if not data:
+        print(f"Could not load file at {filepath}", file=sys.stderr)
+        sys.exit(1)
+
+    return data
+
+
+# Dumps data to a yaml file
+def _dump_config(filepath, data):
+    os.makedirs(os.path.dirname(filepath), exist_ok=True)
```

Comment:
Can you use `pathlib` high-level APIs here (and elsewhere where `os.path` is used)? I think it's more readable and simple.

Reply:
https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module

Reply:
You can convert str to be Path object where it enters the code. For this line specific, it would just be something like `Path(filepath).parent.mkdir(parents=True, exist_ok=True)`. I think this is a more OOP way instead of utilizing the low-level operating system's API.

Reply:
Got it, that worked. Thanks!